lib/console.js

```diff
@@ -249,6 +268,104 @@ Console.prototype.groupEnd = function groupEnd() {
     this[kGroupIndent].slice(0, this[kGroupIndent].length - 2);
 };
 
+const keyKey = 'Key';
+const valuesKey = 'Values';
+const indexKey = '(index)';
+const iterKey = '(iteration index)';
+
+
+const isArray = (v) => ArrayIsArray(v) || isTypedArray(v) || isBuffer(v);
+const inspect = (v) => {
+  const opt = { depth: 0, maxArrayLength: 3 };
+  if (v !== null && typeof v === 'object' &&
+      !isArray(v) && ObjectKeys(v).length > 2)
+    opt.depth = -1;
+  return util.inspect(v, opt);
+};
+
+const getIndexArray = (n) => ArrayFrom({ length: n }, (_, i) => inspect(i));
+
+// https://console.spec.whatwg.org/#table
+Console.prototype.table = function(tabularData, properties) {
+  if (properties !== undefined && !ArrayIsArray(properties)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'properties',
+                               'array', properties);
+  }
+
+  if (tabularData == null ||
+      (typeof tabularData !== 'object' && typeof tabularData !== 'function') ||
+      (!isArray(tabularData) && ObjectKeys(tabularData) > 20))
+    return this.log(tabularData);
+
+  const final = (k, v) => this.log(cliTable(k, v));
+
+  const mapIter = isMapIterator(tabularData);
+  if (mapIter)
+    tabularData = previewMapIterator(tabularData);
+
+  if (mapIter || isMap(tabularData)) {
+    const keys = [];
+    const values = [];
+    for (const [k, v] of tabularData) {
+      keys.push(inspect(k));
+      values.push(inspect(v));
+    }
+    return final([
+      iterKey, keyKey, valuesKey
+    ], [
+      getIndexArray(tabularData.length || tabularData.size),
+      keys,
+      values,
+    ]);
+  }
+
+  const setIter = isSetIterator(tabularData);
+  if (setIter)
+    tabularData = previewSetIterator(tabularData);
+
+  if (setIter || isSet(tabularData)) {
+    const values = [];
+    for (const v of tabularData)
+      values.push(inspect(v));
+    return final([iterKey, valuesKey], [
+      getIndexArray(tabularData.length || tabularData.size),
+      values,
+    ]);
+  }
+
+  const map = {};
+  let hasPrimitives = false;
+  const valuesKeyArray = [];
+  const indexKeyArray = ObjectKeys(tabularData);
+
+  for (var i = 0; i < indexKeyArray.length; i++) {
+    const item = tabularData[indexKeyArray[i]];
+    if (properties === undefined && (item === null ||
+        (typeof item !== 'function' && typeof item !== 'object'))) {
+      hasPrimitives = true;
+      valuesKeyArray[i] = inspect(item);
+    } else {
+      const keys = properties || ObjectKeys(item);
+      for (const key of keys) {
```

Comment:
Suggestion: add a special case in case no keys exist from the item and to show the item in the `Values` in that case.

Reply:
i assume you mean for non-arrays and non-objects?

Reply:
No, I mean e.g. an empty array (`[]`) or empty object (`{}`) etc. Primitives are already handled above, so there is no point for those. That way it this will be visible instead of being invisible.

Another point: this will not work with Set / Map / SetIterator / MapIterator.